lib/lite_cable/iodine.rb

```diff
@@ -0,0 +1,179 @@
+# coding: utf-8
+# frozen_string_literal: true
+
+# TODO:
+# 1. broadcast работает только изнутри сервера Iodine(т.е. если хочется его из экшенов
+# синатры, а не из канала, то надо синатру и каналы запускать сервером iodine).
+# Заработает снаружи, если сделать что-то из этого:
+# a) сделать http путь типо /broadcast изнутри iodine rack.
+# b) запустить и синатру и каналы через сервер iodine(т.е. пуму убрать)
+# c) юзать Redis как pubsub engine И сделать доп. фикс -
+# или делать руками PUBLISH в redis,
+# или если Iodine научится запускаться без запуска сервера, как клиент к pubsub'у.
+#
+# 2. какие-то траблы с памятью при использовании одновременно с синатрой
+# https://github.com/boazsegev/iodine/issues/27
+#
+# 3. puts не работает внутри rack и сообщений с вебсокетов.
+
+module LiteCable # :nodoc:
+  # Iodine extensions
+  module Iodine
+    require "iodine"
+    # LiteCable::Server needs for subscribers_map -
+    # because Iodine don't have unsubscribe_from_all feature
```

Comment:
I guess, using `subscribers_map` and _manually_ (i.e. within MRI) doing broadcasts we lose most of the Iodine power.

What about the following:
- subscribe sockets to Iodine with prefixed streams (`"<identifier>__<stream>"`)
- keep global identifiers -> streams registry
- use 2-level broadcast: first, to the global stream (like you do now), then to the _identified_ streams

Thus we can move almost all hard work to Iodine.

For `unsubscribe_from_all` we can simply store all the subscriptions in the socket object.

Reply:
As I understood, socket.write in Iodine supposed to be an async method and returns immediately. So it will do a relatively small piece of work within MRI - one loop on sockets, and all data transfers will be handled by Iodine.

I'll also ask about this the Iodine creator to be sure

Reply:
Ok, sounds good 👍